src/rest_api/routing.clj

```diff
@@ -0,0 +1,104 @@
+(ns rest-api.routing
+  (:require 
+   [cheshire.core :as json]
+   [clojure.string :as str]
+   [compojure.api.sweet :as sweet]
+   [datomic.api :as d]
+   [rest-api.db.main :refer [datomic-conn]]
+   [ring.util.response :as ring]
+   [schema.core :as schema]))
+
+(defn- json-response [data]
+  (-> data
+      (json/generate-string {:pretty true})
+      (ring/response)
+      (ring/content-type "application/json")))
+
+(defn- entity-not-found [schema-name id]
+  (-> {:message (format "Entity %s %s does not exist" schema-name id)}
+      (json-response)
+      (ring/status 404)))
+
+(defn- wrap-entity-handler [entity-handler]
+  (fn [schema-name id]
+    (let [db (d/db datomic-conn)
+          attr (keyword schema-name "id")
+          lookup-ref [attr id]]
+      (if-let [wb-entity (d/entity db lookup-ref)]
+        (entity-handler wb-entity)))))
+
+(defn- wrap-entity-handlers [route-spec]
+  (fn [binding]
+    (reduce-kv (fn [m k handler]
+                 (assoc m k (handler binding)))
+               (empty route-spec)
+               route-spec)))
+
+(defmulti create-routes
```

Comment:
The `create-routes` multimethod has 2 x implementation, one for when called for a widget mapping (keyword -> mapping), and one that's called when it's  keyword ->  handler (field).  The prior is implemented identically, after the mapping is wrapped  with `wrap-entity-handlers`.